src/tintx/objects.py

```diff
@@ -262,14 +261,21 @@ def get_object_prop(
                 ygrid_diff,
             )
             fwd = transform.Affine.from_gdal(*geotransform)
-            # use safe dtype in case of num_cells > 255
+
+            # iterate over shapes (we have only two) until
+            # we have the correct one (with num == 1)
+            # see https://github.com/antarcticrainforest/tintX/issues/79
+            def get_shape(shapes: Iterator[tuple]) -> Any:
+                for shp, sid in shapes:
+                    if sid:
+                        return geometry.shape(shp)
+
+            # use safe dtype (uint16) in case of num_cells > 255
             poly.append(
-                geometry.shape(
-                    next(
-                        features.shapes(
-                            (image1 == obj).astype("uint16"), transform=fwd
-                        )
-                    )[0]
+                get_shape(
+                    features.shapes(
+                        (image1 == obj).astype("uint16"), transform=fwd
+                    )
```

Comment:
Nitpick: If I look at this I wonder if you could entirely drop the helper function - and hence don't have to worry about type annotations. How about this:

```
poly += [
    geometry.shape(shp)
    for (shp, sid) in features.shapes((image1 == obj).astype("unit16"), transform=fwd)
    if sid
]
```

Reply:
Thanks, applied.